src/contexts/Configuration/ConfigurationProvider.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState, useEffect } from 'react'
+
+import routing from 'services/routing'
+
+export const ConfigurationContext = React.createContext()
+
+const ConfigurationProvider = ({ children }) => {
+  const [reload, setReload] = useState(0)
+  const [configuration, setConfiguration] = useState({
+    loading: true,
+    reload: () => setReload(state => state + 1)
+  })
+
+  useEffect(() => {
+    if (localStorage.configuration) {
+      setConfiguration(conf => ({
+        ...conf,
+        loading: false,
+        ...JSON.parse(localStorage.configuration)
+      }))
+    }
+
+    fetch(process.env.REACT_APP_GRAPHQL_ENDPOINT + '/configuration', {
+      headers: { 'prisma-service': routing.getPrismaService() }
+    })
+      .then(response => {
```

Comment:
Could async/await be used here?

Reply:
Unfortunately, no:

The useEffect() argument (our arrow function) expect another function for its return for the cleanup.

If we make our arrow function async, it's actually a syntactic sugar to return a Promise. So here, useEffect() wouldn't be able to execute the cleanup function